driver/bpf/filler_helpers.h

```diff
@@ -859,17 +872,37 @@ static __always_inline int __bpf_val_to_ring(struct filler_data *data,
 
 				if (!data->curarg_already_on_frame) {
 					volatile u16 read_size = dpi_lookahead_size;
+					int rc;
 
 #ifdef BPF_FORBIDS_ZERO_ACCESS
 					if (read_size)
-						if (bpf_probe_read(&data->buf[curoff_bounded],
-								   ((read_size - 1) & SCRATCH_SIZE_HALF) + 1,
-								   (void *)val))
+						if (mem == KERNEL)
+						{
+							rc = bpf_probe_read_kernel(&data->buf[curoff_bounded],
+									  ((read_size - 1) & SCRATCH_SIZE_HALF) + 1,
+									  (void *)val);
+						}
+						else
+						{
+							rc = bpf_probe_read_user(&data->buf[curoff_bounded],
+									  ((read_size - 1) & SCRATCH_SIZE_HALF) + 1,
+									  (void *)val);
+						}
```

Comment:
Can we put a couple of curly braces around this block of code? It would make it easier to see the entire block is executed only is `read_size` is true.

Reply:
Also, maybe we can come up with a macro or function that would make this check internally, I see the same check is done multiple times and makes it a bit harder to follow.

Reply:
@Molter73 I extracted a helper routine.